### ISSUE TITLE
Hanging goroutines after adding multiple `start` requests on a single WebSocket

### ISSUE THREAD

**Issue by peterbroadhurst** [hyperledger/firefly#134]
Seeing the following in goroutine output:

### Thread stuck waiting for close
```
goroutine 133 [chan receive, 7 minutes]:
github.com/hyperledger-labs/firefly/internal/events.(*eventDispatcher).close(0xc0001cc000)
	/firefly/internal/events/event_dispatcher.go:412 +0x11e
github.com/hyperledger-labs/firefly/internal/events.(*subscriptionManager).registerConnection(0xc000434960, 0xe862b8, 0x12c5200, 0xc000618a20, 0x24, 0xc00022e700, 0x0, 0x0)
	/firefly/internal/events/subscription_manager.go:333 +0x25e
github.com/hyperledger-labs/firefly/internal/events.(*boundCallbacks).RegisterConnection(0xc000426330, 0xc000618a20, 0x24, 0xc00022e700, 0x16, 0x1)
	/firefly/internal/events/bound_events_callbacks.go:30 +0x5e
github.com/hyperledger-labs/firefly/internal/events/websockets.(*WebSockets).start(0x12c5200, 0xc000618a20, 0x24, 0xc00049c4d0, 0xc49a20, 0xc00049c4d0)
	/firefly/internal/events/websockets/websockets.go:115 +0x10f
github.com/hyperledger-labs/firefly/internal/events/websockets.(*websocketConnection).handleStart(0xc0003d0380, 0xc00049c4d0, 0x200, 0xc49a20)
	/firefly/internal/events/websockets/websocket_connection.go:222 +0xac
github.com/hyperledger-labs/firefly/internal/events/websockets.(*websocketConnection).receiveLoop(0xc0003d0380)
	/firefly/internal/events/websockets/websocket_connection.go:147 +0x52a
created by github.com/hyperledger-labs/firefly/internal/events/websockets.newConnection
	/firefly/internal/events/websockets/websocket_connection.go:61 +0x20e
```

### Multiple threads waiting in `handleStart` to get the mutex held by 133

```
goroutine 147 [semacquire, 7 minutes]:
sync.runtime_SemacquireMutex(0xc0004349b4, 0x0, 0x1)
	/usr/local/go/src/runtime/sema.go:71 +0x47
sync.(*Mutex).lockSlow(0xc0004349b0)
	/usr/local/go/src/sync/mutex.go:138 +0x105
sync.(*Mutex).Lock(...)
	/usr/local/go/src/sync/mutex.go:81
github.com/hyperledger-labs/firefly/internal/events.(*subscriptionManager).registerConnection(0xc000434960, 0xe862b8, 0x12c5200, 0xc0002eacc0, 0x24, 0xc00022f180, 0x0, 0x0)
	/firefly/internal/events/subscription_manager.go:318 +0x5a9
github.com/hyperledger-labs/firefly/internal/events.(*boundCallbacks).RegisterConnection(0xc000426330, 0xc0002eacc0, 0x24, 0xc00022f180, 0x16, 0x1)
	/firefly/internal/events/bound_events_callbacks.go:30 +0x5e
github.com/hyperledger-labs/firefly/internal/events/websockets.(*WebSockets).start(0x12c5200, 0xc0002eacc0, 0x24, 0xc0004100b0, 0xc49a20, 0xc0004100b0)
	/firefly/internal/events/websockets/websockets.go:115 +0x10f
github.com/hyperledger-labs/firefly/internal/events/websockets.(*websocketConnection).handleStart(0xc000490280, 0xc0004100b0, 0x200, 0xc49a20)
	/firefly/internal/events/websockets/websocket_connection.go:222 +0xac
github.com/hyperledger-labs/firefly/internal/events/websockets.(*websocketConnection).receiveLoop(0xc000490280)
	/firefly/internal/events/websockets/websocket_connection.go:147 +0x52a
created by github.com/hyperledger-labs/firefly/internal/events/websockets.newConnection
	/firefly/internal/events/websockets/websocket_connection.go:61 +0x20e
```

**Comment by peterbroadhurst** [comment]
In the application scenario there are lots of connections opening closing, and I can see this scenario which would leave `eventPoller.closed` never closing out:

```
firefly_core_0_1  | [2021-07-17T23:47:26.503Z] TRACE Received: {"type":"start","autoack":false,"namespace":"default","name":"app"} pid=1 websocket=fc9fea03-ec7e-40c7-8464-711272b4c3a6
firefly_core_0_1  | [2021-07-17T23:47:26.504Z]  INFO Dispatcher closing for conn=fc9fea03-ec7e-40c7-8464-711272b4c3a6 subscription=ee30c392-181f-4ef7-92ec-39a6651ab252 pid=1 role=ed[fc9fea03-ec7e-40c7-8464-711272b4c3a6] sub=ee30c392-181f-4ef7-92ec-39a6651ab252/ff_system:app_orgs
firefly_core_0_1  | [2021-07-17T23:47:26.503Z] DEBUG Dispatcher attempting to become leader pid=1 role=ed[fc9fea03-ec7e-40c7-8464-711272b4c3a6] sub=ee30c392-181f-4ef7-92ec-39a6651ab252/ff_system:app_orgs
firefly_core_0_1  | [2021-07-17T23:47:26.504Z] DEBUG Closed before we became leader pid=1 role=ed[fc9fea03-ec7e-40c7-8464-711272b4c3a6] sub=ee30c392-181f-4ef7-92ec-39a6651ab252/ff_system:app_orgs
firefly_core_0_1  | [2021-07-17T23:47:26.504Z] TRACE Received: {"type":"start"} pid=1 websocket=fc9fea03-ec7e-40c7-8464-711272b4c3a6
```

**Comment by nguyer** [comment]
@peterbroadhurst Was this one fixed in the PRs mentioned above? Can we close this one now?
